crates/json-abi/src/utils.rs

```diff
@@ -72,16 +72,31 @@ pub(crate) fn selector(preimage: &str) -> Selector {
     }
 }
 
-type Ret<T> = alloy_sol_type_parser::Result<(String, Vec<T>, Vec<T>, bool)>;
+pub(crate) type ParseSigTuple<T> = (String, Vec<T>, Vec<T>, bool);
+pub(crate) type ParseSigResult<T> = parser::Result<ParseSigTuple<T>>;
+
+/// Strips `prefix` from `s` before parsing with `parser`. `prefix` must be followed by whitespace.
+pub(crate) fn parse_maybe_prefixed<F: FnOnce(&str) -> R, R>(
+    mut s: &str,
+    prefix: &str,
+    parser: F,
+) -> R {
+    if let Some(stripped) = s.strip_prefix(prefix) {
+        if stripped.starts_with(|c: char| c.is_ascii_whitespace()) {
+            s = stripped.trim_start();
+        }
```

Comment:
if we need to explicitly handle ascii whitespace,
then this should be `trim_start_matches(is_ascii_whitespace)` I think, or just trim_start directly?